functional_tests/scylla_operator/test_functional.py

```diff
@@ -87,6 +93,128 @@ def test_single_operator_image_tag_is_everywhere(db_cluster):
+    values = HelmValues({
+        'exposeOptions': {
+            'nodeService': {'type': 'Headless'},
+            'broadcastOptions': {key: {'type': 'PodIP'} for key in ('nodes', 'clients')},
+        },
+        'developerMode': True,
+        'fullnameOverride': cluster_name,
+        'scyllaImage': {
+            'repository': db_cluster.k8s_cluster.params.get('docker_image'),
+            'tag': db_cluster.k8s_cluster.params.get('scylla_version'),
+        },
+        'agentImage': {'tag': db_cluster.k8s_cluster.params.get('scylla_mgmt_agent_version')},
+        'serviceAccount': {'create': True, 'annotations': {}, 'name': f"{cluster_name}-member"},
+        'serviceMonitor': {'create': False},
+        'datacenter': dc_name,
+        'racks': [{
+            'name': db_cluster.k8s_cluster.params.get('k8s_scylla_rack'),
+            'members': 3, 'storage': {'capacity': '2Gi'},
+            'resources': {key: {'cpu': 1, 'memory': "200Mi"} for key in ('limits', 'requests')}
+        }]
+    })
+
+    create_namespace(db_cluster=db_cluster, namespace=namespace)
+    db_cluster.k8s_cluster.create_scylla_manager_agent_config(namespace=namespace)
+
+    def get_pod_names_and_ips(cluster_name: str, namespace: str):
+        pod_names_and_ips = kubectl(
+            "get pods --no-headers -o=custom-columns=':.metadata.name,:.status.podIP'"
+            f" -l scylla/cluster={cluster_name}",
+            namespace=namespace).stdout.split("\n")
+        pod_names_and_ips = [row.strip() for row in pod_names_and_ips if row.strip()]
+        assert pod_names_and_ips
+        assert len(pod_names_and_ips) == 3
+        pod_data = {namespace: {}}
+        for pod_name_and_ip in pod_names_and_ips:
+            pod_name, pod_ip = pod_name_and_ip.split()
+            pod_data[namespace][pod_name.strip()] = pod_ip.strip()
+            assert pod_ip not in ("", "None"), "Pod IPs were expected to be set"
+        return pod_data
+
+    log.info('Deploy first ScyllaCluster')
+    try:
+        log.debug(db_cluster.k8s_cluster.helm_install(
+            target_chart_name=target_chart_name, source_chart_name="scylla-operator/scylla",
+            version=operator_version, use_devel=True, values=values, namespace=namespace))
+        db_cluster.k8s_cluster.kubectl_wait("--all --for=condition=Ready pod", namespace=namespace, timeout=1200)
+
+        # NOTE: check that Scylla services are headless - no IPs are set
+        svc_ips = kubectl(
+            "get svc --no-headers -o=custom-columns=':.spec.clusterIP'"
+            f" -l scylla/cluster={cluster_name} -l scylla-operator.scylladb.com/scylla-service-type=member",
+            namespace=namespace).stdout.split()
+        assert svc_ips
+        assert len(svc_ips) == 3
+        assert all(svc_ip in ('', 'None') for svc_ip in svc_ips), "SVC IPs were expected to be absent"
+
+        # NOTE: read Scylla pods IPs
+        pods_data = get_pod_names_and_ips(cluster_name=cluster_name, namespace=namespace)
```

Comment:
`namespace_pod_to_ip_map`?

Reply:
`namespace_pod_to_ip_map` is not really correct if follow this logic.
It may be, using this logic, following: `namespace_pod_name_to_ip_map`.
But it is still ambiguous. Name doesn't mean we have hierarchy.
So, I tend to leave it `as is`.

Reply:
both are fine, it's a local function in the test, and hopefully won't be using it in any other place